nornir/core/connections.py

```diff
@@ -23,7 +25,7 @@ class ConnectionPlugin(ABC):
     __slots__ = ("connection", "state")
 
     def __init__(self) -> None:
-        self.connection: Any = UnestablishedConnection()
+        self.connection: Any = UNESTABLISHED_CONNECTION
```

Comment:
why not `None`? Looking at the code it feels it'd serve the same purposes and simplify one of the `if` statements as you end up checking for `None` and for this.

Reply:
replaced sentinel object with None